lib/cache/cache.go

```diff
@@ -1019,11 +1067,13 @@ func (c *Cache) notify(ctx context.Context, event Event) {
 //	we assume that this cache will eventually end up in a correct state
 //	potentially lagging behind the state of the database.
 func (c *Cache) fetchAndWatch(ctx context.Context, retry retryutils.Retry, timer *time.Timer) error {
+	requestKinds := c.watchKinds()
 	watcher, err := c.Events.NewWatcher(c.ctx, types.Watch{
-		QueueSize:       c.QueueSize,
-		Name:            c.Component,
-		Kinds:           c.watchKinds(),
-		MetricComponent: c.MetricComponent,
+		Name:                c.Component,
+		Kinds:               requestKinds,
+		QueueSize:           c.QueueSize,
+		MetricComponent:     c.MetricComponent,
+		AllowPartialSuccess: true,
```

Comment:
This should be true for every cache *except* the auth server's own cache. An auth server's cache is watching its own local backend impl, so any unimplemented resource there is a bug, and we want to catch it during testing.

Reply:
Great point. Since it should be enabled in most cases, instead of tracking all places that initialize cache, I added the `DisablePartialHealth` field to `cache.Config` and updated `cache.ForAuth` to flip it on: https://github.com/gravitational/teleport/pull/24080/commits/6bed4ae675d74b0484a641d8bdf59418a58f0834